### ISSUE TITLE
Unable to update node to be root

### ISSUE THREAD

**Issue by Machecek** [lazychaser/laravel-nestedset#208]
Hi,

at first, thanks for great package, after using Baum for a while i consider your package as much better.

I have problem with updating node to be parent if he was saved as child. I will try to explain:

1. Created category named "sub" and save it as parent of category named "main"
2. Edited category "sub" and set parent_id = null, but it won't save to DB

```
    $cat = new \App\Category();

    $main = $cat->create([
        'parent_id' => null,
        'title' => 'main'
    ]);

    $sub = $cat->create([
        'parent_id' => $main->id,
        'title' => 'sub'
    ]);

    // All good
    var_dump($cat->get()->toTree());

    $sub->fill([
        'parent_id' => null
    ])->save();

    // Sub is still child of main
    var_dump($cat->get()->toTree());
```

I have also tried $node->saveAsRoot(); and $node->makeRoot()->save(); but it did not work either.

I'm using dimsav/laravel-translatable package (previously with etrepat/baum it works), but i'm struggling with saving parent_id after switching to your package. 

I know that there were some issues between your package and laravel-translatable, i have tried overwrite translatable package as @freekmurze described at https://github.com/dimsav/laravel-translatable/issues/25#issuecomment-47740434, and also @michaelstivala at https://github.com/dimsav/laravel-translatable/issues/25#issuecomment-156105196 but with no success.

All the rest functionality seems to be working properly except this small issue with parent_id. Do you have any idea how to overcome this issue?

Many thanks

========
Laravel version: 5.2.45

**Comment by lazychaser** [comment]
Hi, does it work when you remove translatable?

**Comment by Machecek** [comment]
It's working properly

**Comment by mgsmus** [comment]
I have same problem, @michaelstivala solution didn't work either

**Comment by lazychaser** [comment]
Can't help you, sorry. Issue is caused by other package

**Comment by mgsmus** [comment]
@lazychaser Thanks for reply. I'll try to find a solution tonight. If I find a smooth fix, I'll also post here .

**Comment by lazychaser** [comment]
@mgsmus I would be gratefull for a pull request that fixes this issue

**Comment by mgsmus** [comment]
Ok I found something. 
```
{
  "id": 3,
  "title": "Test Item",
  "_lft": 4,
  "_rgt": 5,
  "parent_id": 1,
}
```

```
$item = Item::find(3);
$item->parent_id = null;
return $item->getDirty();
```

It should return 

```
{
  "parent_id": null
}
```

but it returns an empty array instead, so there are no dirty attributes at all. Translatable package looks dirty attributes count for parent::save() but because of there are no dirty attributes, it doesn't call parent::save, nestedset does not fire save event. If you ask me, if parent id changed, it should also be dirty, shouldn't it? @lazychaser 

Sorry, I can't make a pull request right now, I haven't found a solution to make parent id dirty in your package yet.

**Comment by lazychaser** [comment]
Fixed in v4.2.5
